src/proto/beats.proto

```diff
@@ -10,22 +11,75 @@ enum Source {
   USER = 2;
 }
 
-message Beat {
-  optional int32 frame_position = 1;
-  optional bool enabled = 2 [ default = true ];
-  optional Source source = 3 [ default = ANALYZER ];
+enum SectionType {
+  INTRO = 0;
+  OUTRO = 1;
+  OTHER = 2;
+}
+
+// Setting grey as the default color
+message Color {
+  optional int32 red = 1 [ default = 100 ];
+  optional int32 green = 2 [ default = 100 ];
+  optional int32 blue = 3 [ default = 100 ];
 }
 
 message Bpm {
-  optional double bpm = 1;
-  optional Source source = 2 [ default = ANALYZER ];
+  optional double bpm = 1 [ default = 0 ];
+  optional Source source = 2;
 }
 
-message BeatMap {
-  repeated Beat beat = 1;
+message BpmMarker {
+  optional int32 beat_index = 1;
+  optional Bpm bpm = 2;
 }
 
-message BeatGrid {
-  optional Bpm bpm = 1;
-  optional Beat first_beat = 2;
+message TimeSignature {
+  optional int32 beats_per_bar = 1 [ default = 4];
```

Comment:
I agree that this is the only exception where special default values are acceptable.